### PR TITLE
[HDF5] PointSetOutput Updates

### PR DESCRIPTION
**Pull request by matekelemen**
- fix the ```HDF5File``` context portability issues and make it create the parent directory structure if it doesn't exist yet.
- make the parameters used in python tests explicit so they can be used as examples

### CODE REVIEW

**Comment by matekelemen** [comment]
@sunethwarna ready for review

File: applications/HDF5Application/custom_io/hdf5_file.cpp [hunk 1]
```diff
@@ -191,7 +191,9 @@ File& File::operator=(File&& rOther)
 
 File::~File()
 {
-    H5Fclose(m_file_id);
+    if (0 <= m_file_id) {
+        H5Fclose(m_file_id);
+    }
```
Comment (sunethwarna) [inline applications/HDF5Application/custom_io/hdf5_file.cpp:196]: I suppose you have already checked this with the documentation ryt? There is no anyother way given by hdf5 to check whether the handle is valid?

File: applications/HDF5Application/custom_io/hdf5_file.cpp [hunk 2]
```diff
@@ -629,6 +631,17 @@ void File::Flush()
         << "H5Fflush failed." << std::endl;
 }
 
+void File::Close()
+{
+    if (0 <= m_file_id) {
+        const auto close_result = H5Fclose(m_file_id);
+        KRATOS_ERROR_IF(close_result < 0) << "Failed to close " << m_file_name << " with error code " << close_result;
+        m_file_id = -1;
```
Comment (sunethwarna) [inline applications/HDF5Application/custom_io/hdf5_file.cpp:639]: I suppose again hdf5 is not using the negative values.

File: applications/HDF5Application/python_scripts/core/file_io.py [hunk 3]
```diff
@@ -150,3 +150,24 @@ def Create(settings, data_comm):
...
+        if self.__parameters.Has("io_type"):
+            self.__parameters["io_type"].SetString(distributed)
+        else:
+            self.__parameters.AddString("io_type", distributed)
+        self.__file = Create(self.__parameters, self.__model_part.GetCommunicator().GetDataCommunicator()).Get(self.__model_part)
+        return self.__file
+
+    def __exit__(self, exit_type, exit_value, exit_traceback) -> None:
+        self.__file.Close()
+        self.__file = None
```
Comment (sunethwarna) [inline applications/HDF5Application/python_scripts/core/file_io.py:173]: We do not need `self.__file = None` anymore ryt? the objects will automatically be destroyed when exiting the context.

**Review by sunethwarna — COMMENTED** [review]
This is fine by me.. I only have minor comments.

File: applications/HDF5Application/custom_io/hdf5_file.cpp (hunk 1 above)
Comment (matekelemen) [reply applications/HDF5Application/custom_io/hdf5_file.cpp:196]: I checked the doc and haven't found any exposed function that'd check whether the file is open, but it fails if it's closed twice.

File: applications/HDF5Application/custom_io/hdf5_file.cpp (hunk 2 above)
Comment (matekelemen) [reply applications/HDF5Application/custom_io/hdf5_file.cpp:639]: negative values are reserved for error codes, and ```m_file_id``` defaults to ```-1``` so it should be safe.

File: applications/HDF5Application/python_scripts/core/file_io.py (hunk 3 above)
Comment (matekelemen) [reply applications/HDF5Application/python_scripts/core/file_io.py:173]: True, it's not really necessary; I'm just setting it so it's impossible to abuse the context. There's no way to check whether ```HDF5::File``` is in a valid state right now, so I'm setting ```__file``` to ```None``` whenever it's in an invalid state.

**Review by sunethwarna — APPROVED** [review]

**Pull request merged**